app/views/Settings.js

```diff
@@ -40,12 +62,36 @@ class SettingsScreen extends Component {
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+    GetStoreData(PARTICIPATE)
+      .then(isParticipating => {
+        if (isParticipating === 'true') {
+          this.setState({
+            isLogging: true,
+          });
+          this.willParticipate();
+        } else {
+          this.setState({
+            isLogging: false,
+          });
+        }
+      })
+      .catch(error => console.log(error));
   }
 
   componentWillUnmount() {
     BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
   }
 
+  locationToggleButtonPressed() {
+    this.state.isLogging ? LocationServices.stop() : LocationServices.start();
```

Comment:
If this failed, then what happens to the state stored in`PARTICIPATE`? Not the responsibility of services, hence why I recommend this is a service or hook.